source/agora/common/DNS.d

```diff
@@ -722,16 +722,43 @@ public struct ResourceRecord
         {
             this.uri = val;
         }
+    }
+
+    public RDATA rdata;
 
-        public void toString (scope void delegate(in char[]) @safe sink)
-            const scope @trusted
+    /// Pretty-print the record
+    public void toString (scope void delegate(in char[]) @safe sink)
+        const scope @trusted
+    {
+        switch (this.type)
         {
-            formattedWrite!"%u byte(s)"(sink, this.binary.length);
+        case TYPE.OPT:
+            formattedWrite!"OPT (payload: %s)"(
+                sink, cast(ushort) this.class_);
+            break;
+        case TYPE.A:
+            formattedWrite!"name: %s, TYPE: %s, RDATA: %s"(
+                sink, this.name, this.type, this.rdata.a);
```

Comment:
We could've used https://dlang.org/phobos/std_socket.html#.InternetAddress.addrToString for a better output here